### PR TITLE
Migrate monitor resource to python

### PR DESCRIPTION
**Pull request by zippolyte**
SSIA

Also removes the monitor state from the resource schema, as it can dynamically change between api calls, so it cannot be useful in the resource state.

### CODE REVIEW

File: datadog-monitors-monitor-handler/docs/monitorstate-groups.md
Comment (nmuesch) [inline datadog-monitors-monitor-handler/docs/monitorstate-groups.md:1]: Are these changed as part of the resource migration, or just the doc generation wasn't up to date?

File: datadog-monitors-monitor-handler/example_inputs/inputs_1_create.json
Comment (nmuesch) [inline datadog-monitors-monitor-handler/example_inputs/inputs_1_create.json:1]: I think this can be removed? IIUC its an example input for the tests based on the template schema?

File: datadog-monitors-monitor-handler/src/datadog_monitors_monitor/version.py
Comment (nmuesch) [inline datadog-monitors-monitor-handler/src/datadog_monitors_monitor/version.py:1]: where is this used? 
Ahh is it for the telemetry?

File: datadog-monitors-monitor-handler/src/datadog_monitors_monitor/handlers.py
Comment (nmuesch) [inline datadog-monitors-monitor-handler/src/datadog_monitors_monitor/handlers.py:71]:
```suggestion
        # metric alert and query alert are interchangeable, so don't update from one to the other
```

**Review by nmuesch — COMMENTED** [review]
Left an initial review with a few inline comments, will take a deeper pass tomorrow!

File: datadog-monitors-monitor-handler/src/datadog_monitors_monitor/handlers.py
Comment (nmuesch) [inline datadog-monitors-monitor-handler/src/datadog_monitors_monitor/handlers.py:241]: I believe we still have to call the read handler here, otherwise you get back a model that doesn't include the read only attributes. (Same for the update_handler)

```suggestion
    return read_handler(session, request, callback_context)
```

File: datadog-monitors-monitor-handler/docs/monitorstate-groups.md
Comment (zippolyte) [reply datadog-monitors-monitor-handler/docs/monitorstate-groups.md:1]: I think doc wasn't up to date with model

File: datadog-monitors-monitor-handler/src/datadog_monitors_monitor/version.py
Comment (zippolyte) [reply datadog-monitors-monitor-handler/src/datadog_monitors_monitor/version.py:1]: Yes telemetry

Comment (zippolyte) [reply datadog-monitors-monitor-handler/src/datadog_monitors_monitor/version.py:1]:
```suggestion
__version__ = "2.2.0"
```

File: datadog-monitors-monitor-handler/src/datadog_monitors_monitor/handlers.py
Comment (zippolyte) [reply datadog-monitors-monitor-handler/src/datadog_monitors_monitor/handlers.py:241]: true, wanted to do it, forgot 😅

**Review by nmuesch — APPROVED** [review]
👍 LGTM, lets merge once the common package is merged (Also the safe access of optional non-nullable fields we discussed offline :) )

File: datadog-monitors-monitor-handler/requirements.txt
Comment (nmuesch) [inline datadog-monitors-monitor-handler/requirements.txt:1]: Should we just have this be required in the common package?

File: datadog-monitors-monitor-handler/src/datadog_monitors_monitor/version.py
Comment (nmuesch) [inline datadog-monitors-monitor-handler/src/datadog_monitors_monitor/version.py:1]: Can we do a beta.1 ?

File: datadog-monitors-monitor-handler/requirements.txt
Comment (zippolyte) [reply datadog-monitors-monitor-handler/requirements.txt:1]: Probably yeah, let's do in a next PR, once it's done in the common package ?

File: datadog-monitors-monitor-handler/src/datadog_monitors_monitor/version.py
Comment (zippolyte) [reply datadog-monitors-monitor-handler/src/datadog_monitors_monitor/version.py:1]: Actually this should even be
```suggestion
__version__ = "2.2.0.dev"
```

File: datadog-monitors-monitor-handler/src/datadog_monitors_monitor/handlers.py
Comment (nmuesch) [inline datadog-monitors-monitor-handler/src/datadog_monitors_monitor/handlers.py:49]: Is it possible to set a default in the schema?

Comment (zippolyte) [reply datadog-monitors-monitor-handler/src/datadog_monitors_monitor/handlers.py:49]: Sure, let's do it in a next PR, in case we do it in the common package too

**Review by nmuesch — APPROVED** [review]

**Pull request merged**
